File: codes/cpp/Code_3_01_1.cpp
bool isprime(long long N) {
	// N を 2 以上の整数とし、N が素数であれば true、素数でなければ false を返す
	for (long long i = 2; i <= N - 1; i++) {
		// N が i で割り切れた場合、この時点で素数ではないと分かる
		if (N % i == 0) return false;
	}
	return true;
}

File: codes/python/Code_3_01_1.py
# 2 以上の整数 N に対し、N が素数であれば true、素数でなければ false を返す関数
def isprime(N):
	for i in range(2, N):
		if N % i == 0:
			return False # N が i で割り切れた場合、この時点で素数ではないと分かる
	return True

N = int(input())
if isprime(N):
	print("prime")
else:
	print("not prime")
